### PR TITLE
Add custom volume name support for Gluster Block Volumes.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 1]
```diff
@@ -188,6 +193,11 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 	if cfg.opMode == glusterBlockOpmode {
 		blockVolName = blockVolPrefix + string(uuid.NewUUID())
 	}
+
+	if cfg.volumeNamePrefix != "" {
+		blockVolName = fmt.Sprintf("%s_%s_%s_%s", cfg.volumeNamePrefix, p.options.PVC.Namespace, p.options.PVC.Name, uuid.NewUUID())
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: this blockVolName is not used; it is reset

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 2]
```diff
@@ -204,13 +214,19 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
...
+			iscsiVol.BlockVolName = blockVolPrefix + blockVol.heketiBlockVolRes.ID
+		}
+		iscsiVol.VolumeID = blockVol.heketiBlockVolRes.ID
 	} else if cfg.opMode == glusterBlockOpmode && blockVol.glusterBlockExecVolRes != nil {
 		iscsiVol.Portals = blockVol.glusterBlockExecVolRes.Portals
 		iscsiVol.Iqn = blockVol.glusterBlockExecVolRes.Iqn
 		iscsiVol.User = blockVol.glusterBlockExecVolRes.User
 		iscsiVol.AuthKey = blockVol.glusterBlockExecVolRes.AuthKey
 		iscsiVol.BlockVolName = blockVolName
+		iscsiVol.VolumeID = ""
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: why empty?

Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: @rootfs in gluster-block mode, this volID has no significance, so the annotation has set to ""

**Comment by humblec** [comment]
@childsb Can you please review/merge this ?

**Pull request merged**